### ISSUE TITLE
src/log.c: resource leak suspected by coverity

### ISSUE THREAD

**Issue by chipitsine** [haproxy/haproxy#926]
```
921
    17. Condition p != end, taking true branch.
    25. Condition p != end, taking true branch.
 922                while (p != end) {
    18. Condition !get_logsrv_smp_range(&low, &high, &p, err), taking false branch.
    26. Condition !get_logsrv_smp_range(&low, &high, &p, err), taking true branch.
 923                        if (!get_logsrv_smp_range(&low, &high, &p, err))
    27. Jumping to label error.
    CID 1401272 (#4 of 4): Resource leak (RESOURCE_LEAK)28. leaked_storage: Variable smp_rgs going out of scope leaks the storage it points to.
 924                                goto error;
```

**Comment by wtarreau** [comment]
That's a valid one, fix to be backported as far as 2.0. And there was another leak in it, logsrv->ring_name. Anyway the impact is null for now because in case of error we quit :-)

**Comment by capflam** [comment]
Now backported to all versions. Thanks !
